dlt/dlt_broker_handlers.py

```diff
@@ -199,6 +200,126 @@ def stop(self):
             self.join()
 
 
+class DLTFileSpinner(Process):
```

Comment:
`DLTFileSpinner` copy a lot of code from `DLTMessageHandler`. It's not a good habit. Please think about how to share the same implementation without inheritance.

I don't see any big difference between `DLTFileSpinner` and `DLTMessageHandler` except `self._client` and `py_dlt_client_main_loop`.

There are several ways to refine it. please refactor it.

hint: 

1. Object-Oriented Programming: Dependency Injection, Strategy Pattern and Data Object.
2. Functional Programming: type class (Haskell), Traits (Rust), Concept (C++20), Protocol (Python)

Reply:
Without introducing too many fancy mechanics, simply inheriting from DLTMessageHandler and overriding the different methods should work, no?

Reply:
inheriting from DLTMessageHandler, then we have to pass None for these unnecessary variables we dont need to DLTMessageHandler, and we also dont need self._client_connect(). If class A has more than class B needs, then I dont think class A is a good candidate as a base class.

Sure, current approach is ugly, and inheritance from DLTMessageHandler is equally ugly, why not inheritance? well, i simply pick up the ugly way I want, if the other way does not make it better :-p

Reply:
@lzhou3957 you don't need DLTFileSpinner at all. It is a type of DLTMessageHandler. To keep things simple and still maintain OOP principles, you need to create a base DLTMessagehandler and from that you get two DLTMessageHandler classes one which deals with DLTClient and one that deals with DLTFile. The rest of the same code is shared between the two in the base class. Then using the OOP methodology you instantiate the DLTMessagehandler correspondingly based on the arguments passed to DLTBroker. If there's no IP passed but filepath passed, then you use the File based message handler, otherwise you use DLTClient based message handler as it is written already. This allows us future proof design if we get any other kind of message handling, we can do that easily. DLTBroker doesn't need to be changed much as well. You keep backwards compatibility too. Some technical ways to achieve this have been told by @yen3 already. If you still have concerns we can have a discussion about how to best achieve this. Alot of other unnecessary code changes that you have needed to do, will also go away.

Reply:
sorry, i dont want to touch the other branch for DLTConnector with DLTClient at this moment.
The common base class is already considered to be handled in the second ticket.